application/F3DColorMapTools.cxx

```diff
@@ -0,0 +1,71 @@
+#include "F3DColorMapTools.h"
+
+#include "image.h"
+#include "log.h"
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+namespace F3DColorMapTools
+{
+std::string Find(const std::string& str)
+{
+  if (fs::exists(str))
+  {
+    // already full path;
+    return str;
+  }
+
+  // todo: stem, find file in standard paths
+
+  return {};
+}
+
+std::vector<double> Read(const std::string& path)
+{
+  try
+  {
+    f3d::image img(path);
+
+    if (img.getChannelCount() < 3)
+    {
+      f3d::log::error("The specified color map must have at least 3 channels");
+      return {};
+    }
+
+    if (img.getWidth() < 1)
+    {
+      f3d::log::error("The specified color map width is invalid");
+      return {};
+    }
+
+    if (img.getHeight() != 1)
+    {
+      f3d::log::warn("The specified color map height is not equal to 1, only the first row is "
+                     "taken into account");
+    }
+
+    int w = img.getWidth();
+
+    std::vector<double> cm;
+    cm.resize(4 * w);
+
+    for (int i = 0; i < w; i++)
+    {
+      cm[4 * i] = static_cast<double>(i) / w;
+      for (int j = 0; j < 3; j++)
+      {
+        cm[4 * i + j + 1] = img(i, 0, j);
+      }
+    }
```

Comment:
scale the pixel values to `[0,1]` here instead of within `image::operator(x,y,c)` ?
```suggestion
    for (int i = 0; i < w; i++)
    {
      cm[4 * i] = static_cast<double>(i) / (w - 1);
      for (int j = 0; j < 3; j++)
      {
        cm[4 * i + j + 1] = img(i, 0, j) / max;
      }
    }
```

with the channel type switch just above the loop
```cpp
    double max = 1.0;
    switch (img.getChannelType())
    {
      case f3d::image::ChannelType::BYTE:
        max = 255.0;
        break;
      case f3d::image::ChannelType::SHORT:
        max = 65535.0;
        break;
      default:
        break;
    }
```

and for some extra subjective and inconsequential readability considerations:

```suggestion
    constexpr int y = 0;
    for (int x = 0; x < w; ++x)
    {
      cm[4 * x + 0] = static_cast<double>(x) / (w - 1);
      cm[4 * x + 1] = img(x, y, 0) / max;
      cm[4 * x + 2] = img(x, y, 1) / max;
      cm[4 * x + 3] = img(x, y, 2) / max;
    }
```